### PR TITLE
[RDY] Project documentation

### PR DESCRIPTION
**Pull request by stsewd**
Hi, this is an initial proposal to add the project documentation using sphinx, later we could host the docs on https://readthedocs.org/ so it gets updated on each commit.

This is an example of how it looks

screenshot-2018-2-3 neovim class neovim documentation

Please let me know if you are agree with this, so I can keep with this PR :). Thanks for so awesome API!

### CODE REVIEW

**Comment by justinmk** [comment]
Nice. Can you write the prerequisites needed to run this? Maybe it should be added to CI ?

**Comment by stsewd** [comment]
@justinmk sorry for the late reply, I added the docs env to tox. I'll be writing the docs on the next days/weeks :)

File: docs/index.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Neovim's Python Client
-======================
+Neovims Python Client
```
[the remark this answers is not in the document]
Comment (stsewd) [reply docs/index.rst]: Right now I fixed it :smile:

**Comment by stsewd** [comment]
Please feel free to correct me any grammatical error or misspelling, since English isn't my native language.

**Comment by justinmk** [comment]
LGTM. I don't think we need 19 commits of granularity, I'll squash to 1 unless there are a couple changes you think need to be separated in the history.

Is this still WIP or ready to go?

File: docs/api/nvim.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Neovim Class
```
Comment (justinmk) [inline docs/api/nvim.rst]: shouldn't this be "Nvim Class"?

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#
+# source_suffix = ['.rst', '.md']
+source_suffix = '.rst'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = 'Neovim Python Client'
+copyright = '2018, Neovim'
```
Comment (justinmk) [inline docs/conf.py]: probably should be 2014

**Comment by stsewd** [comment]
> LGTM. I don't think we need 19 commits of granularity, I'll squash to 1 unless there are a couple changes you think need to be separated in the history.

Yeah, one commit is fine for me too (sorry, I do every little separate change a commit).

> Is this still WIP or ready to go?

Yes, it is still WIP, just need to finish two or three sections.

> What happens next? Do we need to register on RTD website (hope not), or will it pick up the .readthedocs.yml somehow?

Yes, someone with access to the repository must be registered on RTD (if not, you can't configure the webhook), then:

- Link the github account to RTD (optional)
- Import the project from the RTD page.
  - Be careful when choosing the name, that would be the url of the docs and can't be changed, if you choose `Neovim Python Client`, the url is `neovim-python-clien.readthedocs.io`.
- Setup the webhook (this is done automatically if you opt for the first step)
- That's all! With every push to master, the docs get updated!

After that we can include the RTD badge on the readme file and probably.

Sorry for the late reply, again.

**Comment by stsewd** [comment]
@justinmk I think we are ready to go now, please let me know any problem and I will try to fix it the faster that I can. Thank guys, I really appreciate all the work you do on Neovim.

**Comment by bfredl** [comment]
I signed up on RTD, and seem to be able to connect the project. Quite a bit of text in README.md` is duplicated, I suppose we want to change it to RTD links once it is up and running

**Comment by bfredl** [comment]
> Be careful when choosing the name, that would be the url of the docs and can't be changed

@justinmk Maybe this is the time to change the name of the library to pynvim?

**Comment by stsewd** [comment]
@bfredl Yeah, I was thinking the same, it's probably better to keep a simpler readme file and just linking from that to RTD. A PR after this one is merged is better (so we can include the rtd badge too!)

File: tox.ini [hunk 4]
```diff
@@ -1,5 +1,7 @@
 [tox]
-envlist = py{27,33,34,35,36}-{asyncio,pyuv},pypy
+envlist =
+  py{27,33,34,35,36}-{asyncio,pyuv},pypy
```
Comment (bfredl) [inline tox.ini]: py33 should be removed

File: docs/installation.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
+Installation
+============
+
+The Neovim Python client supports Python 2.7, and 3.3 or later.
```
Comment (bfredl) [inline docs/installation.rst]: 3.4

**Comment by bfredl** [comment]
LGTM, merged. Thanks for your work!

**Comment by bfredl** [comment]
http://pynvim.readthedocs.io/en/latest/ seems to work.

However, Readthedocs at the same time claims that it did create a GitHub webhook, and that there is no working webhooks, so automatic updates are disabled.

**Comment by justinmk** [comment]
Good call. Could be interesting to do this for the vim docs ....

**Comment by bfredl** [comment]
@justinmk as project admin I think you have to go to the settings page of this repo and somehow accept the readthedocs hook, or you could give me the right to manage hooks for this repo :)

**Comment by stsewd** [comment]
@bfredl Yes, you need rights to create webhooks on the repo to RTD be able to do it, also you could include @justinmk as maintainer on RTD (the webhook setting now needs to be do it manually, just copying a url).

**Comment by justinmk** [comment]
The generated docs should be mentioned in the README.md. In the meantime I added http://pynvim.readthedocs.io/en/latest/ to the repo description.

**Comment by bfredl** [comment]
@stsewd  It should be fixed now, I pushed commit and its seems the hook is triggered.

**Pull request merged**